docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,241 @@
+
+The CMC expects the payment to follow a strict structure that encodes both the **intent** and the **recipient**:
+
+- The **destination account** of the ICP transfer must be the CMC's account with a subaccount derived from the intended controller's principal.
+- The **memo field** must explicitly indicate the intent to create a canister. This can be expressed as:
+  - A legacy 64-bit unsigned integer memo with value `0x41455243` (ASCII `"CREA"`)
+  - Or, for ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`), a memo blob equal to:
+    ```candid
+    "\43\52\45\41\00\00\00\00"
+    ```
+    which represents `"CREA"` padded to 8 bytes.
+
+- **Parameters:**
+  ```candid
+  record {
+    block_index: nat64;              // The ledger block containing the ICP payment
+    controller: principal;           // The principal who will control the new canister
+    settings: opt CanisterSettings;  // Optional settings like controllers, memory limits, etc.
+    subnet_type: opt text;           // Deprecated: legacy subnet selection
+    subnet_selection: opt SubnetSelection; // Preferred subnet selection method
+  }
+  ```
+
+- **Returns:**
+  ```candid
+  variant { Ok: principal; Err: NotifyError }
+  ```
+
+- **Notes:**
+  - The call is idempotent with respect to `block_index`; repeating it returns the same result.
+  - The controller must match the one encoded in the subaccount used for the payment.
+  - Only transactions that follow the exact memo format will be processed successfully.
+
+
+  ### `notify_top_up`
+
+  Tops up an existing canister by minting cycles based on an ICP payment recorded in the ICP Ledger.
+
+  The CMC expects the payment to follow a strict structure that encodes both the **intent** and the **target canister**:
+
+  - The **destination account** of the ICP transfer must be the CMC's account with a subaccount derived from the target canister’s principal.
+  - The **memo field** must explicitly indicate the intent to top up a canister. This can be expressed as:
+    - A legacy 64-bit unsigned integer memo with value `0x50555054` (ASCII `"PUPT"`)
+    - Or, for ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`), a memo blob equal to:
+      ```candid
+      "\50\55\50\54\00\00\00\00"
+      ```
+      which represents `"PUPT"` padded to 8 bytes.
+
+  - **Parameters:**
+    ```candid
+    record {
+      block_index: nat64;     // Block index of the ICP ledger payment
+      canister_id: principal; // Canister to be topped up
+    }
+    ```
+
+  - **Returns:**
+    ```candid
+    variant { Ok: nat; Err: NotifyError } // Number of cycles minted and sent
+    ```
+
+  - **Notes:**
+    - The subaccount used in the transfer must be derived from the target canister’s principal using the standard 32-byte encoding (see “Shared Logic” section).
+    - This method is idempotent by `block_index`: retrying the same request will return the same result if it already succeeded.
```

Comment:
Same here, probably worth noting cache expiry time and that it can only be called on recent-enough blocks